test/unit/node/heart.test.ts

```diff
@@ -0,0 +1,99 @@
+import { readFile, writeFile } from "fs/promises"
+import { logger } from "@coder/logger"
+import { Heart, heartbeatTimer } from "../../../src/node/heart"
+import { clean, mockLogger, tmpdir } from "../../utils/helpers"
+
+function mockIsActive(resolveTo: boolean): () => Promise<boolean> {
+  return () => new Promise((resolve, reject) => setTimeout(() => resolve(resolveTo), 100))
+}
```

Comment:
Couldn't decide if this should be defined in the `describe` block of the `Heart` test suite or outside. By being outside, it's only defined once and easier to read the suite (but the suite only runs once anyway 🤷🏼‍♂️).

Reply:
Outside makes sense to me :+1: 

Could we `return Promise.resolve(resolveTo)`?  Might be nice to avoid an extra 100ms wait each time.  Or the `jest.fn().mockResolvedValue` you use below, that looks pretty dope.

Reply:
I don't know why I didn't think of that lol good call!